### ISSUE TITLE
Default section written before global causing log problems

### ISSUE THREAD

**Issue by void-in** [haproxytech/dataplaneapi#24]
First of all, apologies if the problem is obvious because this is my first experience with the dataplaneapi.

So I want to register a new server to the backend depending upon the current rate of requests. I am able to register a new server in the backend and even through it is showing in the /etc/haproxy/haproxy.cfg, still the server is not handling any requests from the frontend. I have verified that the port is accessible and the server is indeed part of the backends.

```python
import json
import time
import requests
from requests.auth import HTTPBasicAuth

def addServersToBackEnd(backEnd):
    uri = "http://localhost:5555/v1/services/haproxy/configuration/servers"
    version = getVersion(backEnd)
    transactionId = getTransactionId(version)
    params = {
        'backend': backEnd,
        'transaction_id': transactionId
    }
    data = {"name": "Server1", "address": "1.1.1.1", "port": 1234, "check": "enabled", "weight": 100}
    res = requests.post(uri, auth=HTTPBasicAuth('dataplaneapi','redacted'), params = params, json = data)
    commitTransaction = 'http://localhost:5555/v1/services/haproxy/transactions/' + transactionId
    res = requests.put(commitTransaction, auth=HTTPBasicAuth('dataplaneapi', 'redacted'))
    
def getTransactionId(version):
    uri = "http://localhost:5555/v1/services/haproxy/transactions"
    params = {
        'version': version
    }
    res = requests.post(uri, auth=HTTPBasicAuth('dataplaneapi', 'redacted'), params = params)
    data = res.json()
    transactionId = data['id']
    return transactionId

def getVersion(backEnd):
    uri = "http://localhost:5555/v1/services/haproxy/configuration/servers"
    params = {
        'backend': backEnd
    }
    res = requests.get(uri, auth=HTTPBasicAuth('dataplaneapi', 'redacted'), params = params)
    if res.status_code == 200:
        data = res.json()
        version = data['_version']
        return version

uri = 'http://localhost:5555/v1/services/haproxy/stats/native'
params = {
    'type': 'backend',
    'name': 'MyBackEnd'
}
serverAdded = False
while True:
    try:
        res = requests.get(uri, auth=HTTPBasicAuth('dataplaneapi', 'redacted'), params = params)
        if res.status_code == 200:
            data = res.json()
            rate = data[0]['stats'][0]['stats']['rate']
            print("Backend: {0}, Connections:{1}".format(str(params['name']),str(rate)))
            if rate > 100 and not serverAdded:
                 addServersToBackEnd(params['name'])
                 serverAdded = True
            else:
                print("Requests under the limit")
        else:
            print("An error occurred while querying the API. Error code is {0}".format(str(res)))
    except requests.exceptions.RequestException as e:
        print e
    time.sleep(1)
```
After this, the server 1.1.1.1 is successfully added to the backend 'MyBackEnd'. However, the incoming requests are still only served by the servers already configured in the backend before the changes committed by the dataplaneapi.

Thank you very much for reading. Any help would be greatly appreciated.

**Comment by mjuraga** [comment]
If it doesn't show in the .cfg file then it means that the server is not added. You should check responses of requests.post and requests.put in def addServersToBackEnd(backEnd) in case something failed.

**Comment by void-in** [comment]
Thanks a lot for the response. The server is visible in the .cfg file after the changes are made by the dataplaneapi. Also, the commit transaction URL returns success so I believe everything is working fine.

But still the haproxy.log file shows that the connections for the 'MyBackEnd' is still handled by only the two previous servers and not the new one added 1.1.1.1

I will try to use the reload_delay option. Thanks once again.

**Comment by mjuraga** [comment]
Try commiting the transaction with force_reload option. It will automatically reload the HAProxy using your -r parameter. If you do not specify it, the reloads happen every <-d s> (default 5s). So when multiple changes are made, the haproxy process will be reloaded every 5s, not on every change.

In that case you should get a response header Reload-ID, and then you can query /services/haproxy/reloads/<reload-id> endpoint to see if the haproxy reloaded successfully.

**Comment by void-in** [comment]
So I have solved all my problems :) Just a nuisance is left. After dataplane API writes the configuration file haproxy.cfg, the first section is the ```default``` while the second section is ```global```. It is causing problems in the logging. Only when I copy-paste the global section at the top is when logs are being successfully written to /var/log/haproxy for all frontends.

@mjuraga Have you encountered such an issue? Or what might be the possible workarounds? Thanks a lot for your help as always.

**Comment by mjuraga** [comment]
Yes this is an issue we encountered, and are working on it.

**Comment by mjuraga** [comment]
Fixed with e308dd805ebf68a2ca02dc1d64b5bdf5d6d281ed
